Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -508,14 +500,16 @@ class BrowserViewController: UIViewController {
                 self.tabManager.restoreTabs()
             },
             noCallback: { _ in
+                TabMO.removeAll()
                 self.tabManager.addTabAndSelect()
             }
         )
         self.present(alert, animated: true, completion: nil)
     }
 
     fileprivate func canRestoreTabs() -> Bool {
-        return !TabMO.getAll().isEmpty
+        // Make sure there's at least one real tab open
+        return !TabMO.getAll().compactMap({ $0.url }).isEmpty
```

Comment:
Nice improvement to existing logic. There are ways to optimize the query here, but given tabs will probably limited in size, I think this is fine.